cloud_functions/extract_embedding.py

```diff
@@ -0,0 +1,147 @@
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https:#www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, softwar3
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""This function creates embeddings from a given image input.
+This cloud function first saves all necessary components of the DELG model to t
+mp directories from GCP storage buckets and after doing so uses tensorflow to l
+oad and run the saved model on the image.
+"""
+import os
+import tempfile
+import file_path_utils as fp
+from google.cloud import storage
+from file_path_utils import get_user_and_dataset_name, get_photo_name
+import cv2
+import numpy as np
+import tensorflow as tf
+
+REQUIRED_SIGNATURE = 'serving_default'
+REQUIRED_OUTPUT = 'global_descriptor'
+EMBEDDING_BUCKET_NAME = 'embeddings_visualizer_output_bucket'
+EMBEDDINGS_FOLDER_NAME = 'embeddings'
+MAX_RESOLUTION = 800
+
+# SETUP (runs once, when deploying):
+
+# Storage API
+storage_client = storage.Client()
+
+# Create tmp dirs for DELG SavedModel. (Hard-drive is locked, must download all files to /tmp!) 
+os.mkdir('/tmp/saved_model')
+os.mkdir('/tmp/saved_model/variables')
+
+# Download DELG SavedModel files:
+model_bucket = storage_client.bucket('delg_model_bucket')
+
+
```

Comment:
Can you follow this blank line rule:
https://google.github.io/styleguide/pyguide.html#35-blank-lines